lib/collectinfo_analyzer/show_controller.py

```diff
@@ -916,17 +1051,204 @@ def do_sindex(self, line):
+                    "%s SIndex Statistics" % (sindex),
                     stats,
                     self.log_handler.get_cinfo_log_at(timestamp=timestamp),
                     show_total=show_total,
                     title_every_nth=title_every_nth,
                     flip_output=flip_output,
                     timestamp=timestamp,
-                    **self.mods
+                    **self.mods,
                 )
 
 
+@CommandHelp(
+    '"show statistics xdr" is used to display xdr statistics for Aerospike components.'
+)
+class ShowStatisticsXDRController(CollectinfoCommandController):
+    def __init__(self):
+        self.modifiers = set(["like", "for"])
+        self.getter = GetStatisticsController(self.log_handler)
+
+    def _check_ns_stats_and_warn(self, xdr_ns_stats):
+        for ts_stats in xdr_ns_stats.values():
+            for node_stats in ts_stats.values():
+                if not node_stats:
+                    self.logger.warning(
+                        "XDR namespace statistics were introduced in server 5.0 and not added to the collectinfo file until asadm 2.13.0"
+                    )
+                    return
+
+    @CommandHelp(
+        "Displays xdr, xdr datacenter, and xdr namespace statistics.",
+        "  Options:",
+        "    -t           - Set to show total column at the end. It contains node wise sum for statistics.",
+        "    -r           - Repeat output table title and row header after every <terminal width> columns.",
+        "                   [default: False, no repetition]",
+        "    --flip       - Flip output table to show Nodes on Y axis and stats on X axis.",
+    )
+    def _do_default(self, line):
+        self._do_xdr(line[:])
+        self.do_dc(line[:])
+        self.do_namespace(line[:])
+
+    def _do_xdr(self, line):
+        show_total = util.check_arg_and_delete_from_mods(
+            line=line, arg="-t", default=False, modifiers=self.modifiers, mods=self.mods
+        )
+        title_every_nth = util.get_arg_and_delete_from_mods(
+            line=line,
+            arg="-r",
+            return_type=int,
+            default=0,
+            modifiers=self.modifiers,
+            mods=self.mods,
+        )
+
+        flip_output = util.check_arg_and_delete_from_mods(
+            line=line,
+            arg="-flip",
+            default=False,
+            modifiers=self.modifiers,
+            mods=self.mods,
+        ) or util.check_arg_and_delete_from_mods(
+            line=line,
+            arg="--flip",
+            default=False,
+            modifiers=self.modifiers,
+            mods=self.mods,
+        )
+
+        xdr_stats = self.getter.get_xdr()
+
+        for timestamp in xdr_stats.keys():
+            cinfo_log = self.log_handler.get_cinfo_log_at(timestamp=timestamp)
+
+            # There are no XDR level stats for XDR 5. This will not print anything
+            # if it is empty
+            self.view.show_stats(
```

Comment:
Just checking to make sure there isn't a specific view function for this. show_stats() is ok?